### PR TITLE
Adds the missing pages for the HID interface

### PR DESCRIPTION
**Pull request by rachelandrew**
Adds the missing pages. HID already had the main interface page.

Again referencing this PR: https://github.com/WICG/webhid/pull/68/files as the spec is quite sketchy on the details.

Reviewer: @jpmedley

### CODE REVIEW

File: files/en-us/web/api/hid/onconnect/index.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  - API
+  - Property
+  - Reference
+  - onconnect
+  - HID
+browser-compat: api.HID.onconnect
+---
+<div>{{securecontext_header}}{{DefaultAPISidebar("WebHID API")}}</div>
+
+<p class="summary">The <strong><code>onconnect</code></strong> EventHandler of the {{domxref("HID")}} interface is an {{event("Event_Handlers", "Event Handler")}} that processes the events fired when the user agent connects to the HID device.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/hid/onconnect/index.html:14]:
```suggestion
<p class="summary">The <strong><code>onconnect</code></strong> {{event("Event_Handlers", "Event Handler")}} of the {{domxref("HID")}} interface processes the events fired when the user agent connects to the HID device.</p>
```

File: files/en-us/web/api/hid/ondisconnect/index.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  - API
+  - Property
+  - Reference
+  - ondisconnect
+  - HID
+browser-compat: api.HID.ondisconnect
+---
+<div>{{securecontext_header}}{{DefaultAPISidebar("WebHID API")}}</div>
+
+<p class="summary">The <strong><code>ondisconnect</code></strong> EventHandler of the {{domxref("HID")}} interface is an {{event("Event_Handlers", "Event Handler")}} that processes the events when the user agent disconnects from the HID device.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/hid/ondisconnect/index.html:14]:
```suggestion
<p class="summary">The <strong><code>ondisconnect</code></strong> {{event("Event_Handlers", "Event Handler")}} of the {{domxref("HID")}} interface processes the events when the user agent disconnects from the HID device.</p>
```

File: files/en-us/web/api/hid/requestdevice/index.html [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      </dd>
+      <dt><code>usage</code>{{Optional_Inline}}</dt>
+      <dd>An integer representing the usage ID component of the HID usage of the requested device.</dd>
+    </dl>
+  </dd>
+</dl>
+
+<div class="notecard note">
+  <h4>Note:</h4>
+  <p>The device filters are used to narrow the list of devices presented to the user. If no filters are present, all connected devices are shown. When one of more filters are included, a device is included if any filter matches. To match a filter, all of the rules included in that filter must match.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/hid/requestdevice/index.html:44]: >one of more filters

Is this supposed to be 'one *or* more filters'?

File: files/en-us/web/api/hid/requestdevice/index.html [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+<h3 id="Exceptions">Exceptions</h3>
+
+<dl>
+  <dt>{{domxref("DOMException")}} <code>SecurityError</code></dt>
+  <dd>Thrown if the page does not allow access to the HID feature.</dd>
+</dl>
+
+<h2 id="Examples">Examples</h2>
+
+<h3>Matching a device will all four filter rules</h3>
```
Comment (jpmedley) [inline files/en-us/web/api/hid/requestdevice/index.html:60]: What?

File: files/en-us/web/api/hid/requestdevice/index.html [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    if (!device) {
+      console.log("No device was selected.");
+    } else {
+      console.log(`HID: ${device.productName}`);
+    }
+  });</pre>
+
+  <h3>An example with two filters</h3>
+
+  <p>This next example includes two filters, devices will be shown if they match either of these filters.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/hid/requestdevice/index.html:92]:
```suggestion
  <p>This next example includes two filters. Devices will be shown if they match either of these filters.</p>
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/hid/requestdevice/index.html (hunk 3 above)
Comment (rachelandrew) [reply files/en-us/web/api/hid/requestdevice/index.html:44]:
```suggestion
  <p>The device filters are used to narrow the list of devices presented to the user. If no filters are present, all connected devices are shown. When one or more filters are included, a device is included if any filter matches. To match a filter, all of the rules included in that filter must match.</p>
```

File: files/en-us/web/api/hid/requestdevice/index.html (hunk 4 above)
Comment (rachelandrew) [reply files/en-us/web/api/hid/requestdevice/index.html:60]:
```suggestion
<h3>Matching a device with all four filter rules</h3>
```

**Comment by rachelandrew** [comment]
@jpmedley I've resolved everything on this one.

**Review by jpmedley — APPROVED** [review]

**Pull request merged**